File: src/main/java/no/nav/fo/forenkletdeploy/provider/TeamProvider.java
package no.nav.fo.forenkletdeploy.provider;

import no.nav.fo.forenkletdeploy.teams.FOTeam;
import no.nav.fo.forenkletdeploy.teams.Team;
import no.nav.fo.forenkletdeploy.teams.TeamOppfolging;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@Component
public class TeamProvider {

    private final List<Team> teams = new ArrayList<>();

    public TeamProvider() {
        teams.add(new FOTeam());
        teams.add(new TeamOppfolging());
    }

    public List<Team> getTeams() {
        return teams;
    }
}

File: src/main/kotlin/no/nav/fo/forenkletdeploy/provider/TeamProvider.kt
package no.nav.fo.forenkletdeploy.provider

import no.nav.fo.forenkletdeploy.teams.FOTeam
import no.nav.fo.forenkletdeploy.teams.Team
import no.nav.fo.forenkletdeploy.teams.TeamOppfolging
import no.nav.fo.forenkletdeploy.teams.TeamSoknad
import org.springframework.stereotype.Component

import java.util.ArrayList

@Component
class TeamProvider {

    val teams = arrayListOf(
            FOTeam(), TeamOppfolging(), TeamSoknad()
    )
}
